source/parser.hera

```diff
@@ -5816,9 +5816,11 @@ Type
   TypeConditional
 
 TypeBinary
-  TypeUnary ( __ TypeBinaryOp __ TypeUnary )* ->
-    if ($2.length) return $0
-    return $1
+  ( __ TypeBinaryOp __ )?:optionalPrefix TypeUnary:t ( __ TypeBinaryOp __ TypeUnary )*:ops ->
```

Comment:
I find it a bit awkward that the leading `__` is inside the optional part... So only with the prefix does this allow leading whitespace.

But I see TypeUnary has the same issue so maybe this is cleanup for later.